### PR TITLE
Document the pkg: prefix for package targets

### PR DESCRIPTION
**Pull request by michaelpj**
As discussed in https://github.com/haskell/cabal/issues/9009

### CODE REVIEW

**Review by Kleidukos — APPROVED** [review]

File: doc/cabal-commands.rst [hunk 1]
```diff
@@ -191,7 +191,7 @@ Target Forms
 
 A cabal command target can take any of the following forms:
 
--  A package target: ``package``, which specifies that all enabled
+-  A package target: ``[pkg:]package``, which specifies that all enabled
```
Comment (jneira) [inline doc/cabal-commands.rst:194]: without reading the whole thing, do you think is clear with the new version you can use `cabal install my-package` if there is no ambiguities?

**Comment by ulysses4ever** [comment]
@mergify rebase

File: doc/cabal-commands.rst (hunk 1 above)
Comment (michaelpj) [reply doc/cabal-commands.rst:194]: The other entries show "optional syntax" in the same way as this. As to whether that's clear... I managed to understand it, but if we change it we should change the whole section so it's consistent.

**Comment by ulysses4ever** [comment]
Some Windows unhappiness on CI. Trying to restart.

**Comment by ulysses4ever** [comment]
Hmm, that didn't help…

**Comment by gbaz** [comment]
the sporadic failing test could be fixed probably by something like using this https://github.com/haskell/cabal/blob/24934dd714928859cbd21d52eefe343f1a270114/cabal-install/tests/UnitTests/TempTestDir.hs#L43

**Comment by ulysses4ever** [comment]
@mergify rebase

**Comment by ulysses4ever** [comment]
@mergify rebase

**Comment by ulysses4ever** [comment]
@Kleidukos how do you feel about backporting documentation improvements? May be a cheap and nice improvement for 3.10.2. Hopefully won't need manual intervention (fourmolu doesn't touch .rst files).

**Comment by Kleidukos** [comment]
yep that's a good one, thank you

**Comment by Kleidukos** [comment]
@mergifyIO backport 3.10

**Pull request merged**